pkg/stores/proxy/proxy_store.go

```diff
@@ -422,20 +427,33 @@ func (s *Store) Create(apiOp *types.APIRequest, schema *types.APISchema, params
 	}
 
 	name := types.Name(input)
-	ns := types.Namespace(input)
-	if name == "" && input.String("metadata", "generateName") == "" {
-		input.SetNested(schema.ID[0:1]+"-", "metadata", "generatedName")
+	namespace := types.Namespace(input)
+	generateName := input.String("metadata", "generateName")
+
+	if name == "" && generateName == "" {
+		input.SetNested(schema.ID[0:1]+"-", "metadata", "generateName")
 	}
-	if ns == "" && apiOp.Namespace != "" {
-		ns = apiOp.Namespace
-		input.SetNested(ns, "metadata", "namespace")
+
+	if attributes.Namespaced(schema) {
+		if namespace == "" {
+			if apiOp.Namespace == "" {
+				return nil, nil, validation.ErrorCode{
+					Status: http.StatusUnprocessableEntity,
+					Code:   errNamespaceRequired,
+				}
+			}
+
+			namespace = apiOp.Namespace
+		}
+
+		input.SetNested(namespace, "metadata", "namespace")
```

Comment:
nit: Can be simplified a bit to remove 3 nested ifs. (Basically, no need to set the namespace if it's already set).


```suggestion
	if attributes.Namespaced(schema) && namespace == "" {
		if apiOp.Namespace == "" {
			return nil, nil, validation.ErrorCode{
				Status: http.StatusUnprocessableEntity,
				Code:   errNamespaceRequired,
			}
		}

		namespace = apiOp.Namespace
		input.SetNested(namespace, "metadata", "namespace")
```